### ISSUE TITLE
Adding custom commit parser, how?

### ISSUE THREAD

**Issue by claudio-alvaro-wbgames** [python-semantic-release/python-semantic-release#568]
Hello! 

Can I get a bit more information about adding a custom commit parser to a project. 

I created a custom parser function in a python file that works as intended when passing a message. I then attempted to add a custom commit parser to the root of a src-layout python project (intended to become a template repo). I reference the commit parser in the `pyproject.toml` under  `tool.semantic_release.commit_parser = "parser_custom.parse_commit_message"` Unfortunately, it seems like the code assumes that the custom parser should be its own package (it's being loaded using importlib.import_module in semantic_release.settings.current_commit_parser). It fails to load the parser function with the message `Unable to import parser "No module named 'parser_custom'"` whenever I run semantic-release commands.

My hope is that there's a simple way to add the python file with the custom parser to the root of my project (as to not pollute its src code). I'm not so keen on creating a new project/package specifically for a custom parser. 

Any info would be appreciated.

**Comment by bernardcooke53** [comment]
Hey :wave:

Could you please indicate which version of `python-semantic-release` you're using, and how (e.g. via GitHub actions, directly in another CI context, etc).

I was able to get `importlib` doing its job locally with the following (on Ubuntu):

```bash
mkdir test && pushd test
touch main.py import_test.py
cat <<EOF > main.py
def hello_world() -> None:
    print("Hello world")
EOF

cat <<EOF > import_test.py
import importlib

main = importlib.import_module("main")
main.hello_world()
EOF

python3 import_test.py
popd
```

**Comment by claudio-alvaro-wbgames** [comment]
Latest 7.33.2. Testing on Windows 10 locally in a cmd shell within a project venv activated.

**Comment by Klaudikus** [comment]
I created a repro project --> https://github.com/Klaudikus/semantic-release-repro/

Copied the angular commit parser to the root and reference it in the pyproject.toml. Cheers!

**Comment by bernardcooke53** [comment]
The repro project is super helpful - thank you @Klaudikus!

I checked this by editing `.venv/lib/python3.8/site-packages/semantic-release/settings.py` to add a breakpoint after setting up the virtual environment (neat trick when debugging other libraries sometimes) - I think the reason this works is because `importlib.import_module` searches the directory with the module it's called from, so using a REPL from the repo root would work because you'd be in the right directory, while calling that function from within the `semantic-release` package wouldn't find it.

In the debugger I checked `sys.path` and the repo root wasn't there, but after I added the root directory to it there was no issue.

I undid those changes and checked again locally using [PYTHONPATH](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH) to add that directory, which worked for me.

I don't have the setup to test the equivalent on Windows but I believe it's something like the following, from the repo root:

```cmd
set PYTHONPATH=%cd%
semantic-release version
``` 
The equivalent on Ubuntu is
```bash
PYTHONPATH="$(pwd)" semantic-release version
```
Hopefully setting that variable will work for you? I have read that `%cd%` in Windows cmd expands to the current directory :crossed_fingers: 

This probably should be part of the docs for a custom commit parser, but I'm hesitant to go manually fiddling with `PYTHONPATH` in all cases since that could lead to unexpected name collisions (e.g. if I had a file called `os.py` at the repo root, the behaviour of `import os` would change).
